### PR TITLE
feat: integrate external data reader for tree r

### PR DESCRIPTION
**Pull request by cryptonemo**

### CODE REVIEW

File: storage-proofs/src/election_post.rs [hunk 1]
```diff
@@ -481,18 +486,40 @@ mod tests {
...
             assert_eq!(c, true);
 
+            // Open the replica for base layer data reads.
+            let reader = OpenOptions::new()
+                .read(true)
+                .open(&replica_path)
+                .expect("Failed to open replica (base layer data)");
+
+            // Configure an external reader for access to the base layer data.
+            let external_reader = ExternalReader {
```
Comment (dignifiedquire) [inline storage-proofs/src/election_post.rs]: this `ExternalReader` is always the same, I would suggest moving this impl into `.lcmerkle_tree` and just give it the path.

File: storage-proofs/src/merkle.rs [hunk 2]
```diff
@@ -285,33 +288,48 @@ pub fn create_merkle_tree<H: Hasher, U: typenum::Unsigned>(
...
-        // iterator case.
-        H::Domain::try_from_bytes(d).expect("failed to convert node data to domain element")
+    // Open the replica for base layer data reads.
+    let reader = OpenOptions::new()
+        .read(true)
+        .open(replica_path)
+        .expect("Failed to open replica (base layer data)");
+
+    // Configure an external reader for access to the base layer data.
+    let external_reader = ExternalReader {
```
Comment (dignifiedquire) [inline storage-proofs/src/merkle.rs]: could use the new method here

File: storage-proofs/src/stacked/params.rs [hunk 3]
```diff
@@ -362,9 +364,9 @@ impl<H: Hasher, G: Hasher> TemporaryAux<H, G> {
         self.labels.column(column_index)
     }
 
-    // "Compact" will discard all persisted data that is no longer
-    // required and compact the remaining "tree_r_last" merkle tree.
-    pub fn compact(t_aux: TemporaryAux<H, G>) -> Result<()> {
+    // 'delete' will discard all persisted merkle and layer data that
+    // is no longer required.
+    pub fn delete(t_aux: TemporaryAux<H, G>) -> Result<()> {
```
Comment (dignifiedquire) [inline storage-proofs/src/stacked/params.rs]: We still keep part of the cache, so I think calling it `compact` is still the right name

Comment (cryptonemo) [reply storage-proofs/src/stacked/params.rs]: We do, but that method only deletes now. Compaction is done elsewhere, immediately after the trees are built.

Comment (cryptonemo) [reply storage-proofs/src/stacked/params.rs]: :+1: Renamed to `clear_temp`

**Pull request merged**